.github/workflows/test.yml

```diff
@@ -7,11 +7,16 @@ on:
     branches: [ master ]
 
 jobs:
-
+  pre-checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      - name: Ensure SHA pinned actions
+        uses: zgosalvez/github-actions-ensure-sha-pinned-actions@f32435541e24cd6a4700a7f52bb2ec59e80603b1 # v2.1.4
   test:
```

Comment:
Not sure if I should add this pre-check here or maybe in the `release` workflow. I just need to run the Ensure SHA pinned actions check.  I added it in a separated job just to separate the concern .
Please advise if there is a better location for it.
cc: @gionn

Reply:
looks good here, otherwise you get a failure once the pr is already merged

can you rename the new job to a more generic `build` and move here the `- uses: ./.github/actions/pre-commit` step below?